test-suite/src/data_type/int32.rs

```diff
@@ -0,0 +1,119 @@
+use {
+    crate::*,
+    gluesql_core::{data::ValueError, prelude::Payload, prelude::Value::*},
+};
+
+test_case!(int32, async move {
+    run!(
+        "CREATE TABLE Item (
+        field_one INT(32),
+        field_two INT(32),
+    );"
+    );
+    run!("INSERT INTO Item VALUES (1, -1), (-2, 2), (3, 3), (-4, -4);");
+
+    let parse_i32 = |text: &str| -> i32 { text.parse().unwrap() };
+
+    let max: i64 = i32::MAX as i64 + 1_i64;
+    let min: i64 = i32::MIN as i64 - 1_i64;
+    
+
+    test!(
+        Err(ValueError::FailedToParseNumber.into()),
+        &format!(
+            "INSERT INTO Item VALUES ({:?}, {:?})",
+            max,
+            min
+        )
+    );
+
+    // cast i32::MAX+1
+    // this should produce an error! will create a different PR / issue for fixing this.
+    //test!(
+    //    Err(ValueError::LiteralCastToDataTypeFailed(DataType::Int32, max.to_string()).into()),
+    //    &format!("select cast({} as INT(32)) from Item", max)
+    //);
+
+    // cast i32::MIN-1
+    // this also should produce an error, will create a PR for it as well.
+    //test!(
+    //    Err(ValueError::LiteralCastToDataTypeFailed(DataType::Int32, min.to_string()).into()),
+    //    &format!("select cast({} as INT(32)) from Item", min)
+    //);
```

Comment:
Sounds good!
This makes sense and I am also for the decision that you made making this as a different PR.